candle-examples/examples/stable-diffusion-3/main.rs

```diff
@@ -100,87 +113,163 @@ fn run(args: Args) -> Result<()> {
+            api.repo(hf_hub::Repo::model(name.to_string()))
+        };
+        let model_file = sai_repo.get("sd3_medium_incl_clips_t5xxlfp16.safetensors")?;
+        let vb_fp16 = unsafe {
+            candle_nn::VarBuilder::from_mmaped_safetensors(&[&model_file], DType::F16, &device)?
+        };
+
+        let (context, y) = {
+            let vb_fp32 = unsafe {
+                candle_nn::VarBuilder::from_mmaped_safetensors(&[model_file], DType::F32, &device)?
+            };
+            let mut triple = StableDiffusion3TripleClipWithTokenizer::new(
+                vb_fp16.pp("text_encoders"),
+                vb_fp32.pp("text_encoders"),
+            )?;
+            let (context, y) = triple.encode_text_to_embedding(prompt.as_str(), &device)?;
+            let (context_uncond, y_uncond) =
+                triple.encode_text_to_embedding(uncond_prompt.as_str(), &device)?;
+            (
+                Tensor::cat(&[context, context_uncond], 0)?,
+                Tensor::cat(&[y, y_uncond], 0)?,
+            )
+        };
 
-    let img = {
-        let vb_vae = vb_fp16
-            .clone()
-            .rename_f(sd3_vae_vb_rename)
-            .pp("first_stage_model");
-        let autoencoder = build_sd3_vae_autoencoder(vb_vae)?;
+        let x = {
+            let mmdit = MMDiT::new(
+                &MMDiTConfig::sd3_medium(),
+                use_flash_attn,
+                vb_fp16.pp("model.diffusion_model"),
+            )?;
 
-        // Apply TAESD3 scale factor. Seems to be significantly improving the quality of the image.
-        // https://github.com/comfyanonymous/ComfyUI/blob/3c60ecd7a83da43d694e26a77ca6b93106891251/nodes.py#L721-L723
-        autoencoder.decode(&((x.clone() / 1.5305)? + 0.0609)?)?
+            if let Some(seed) = seed {
+                device.set_seed(seed)?;
+            }
+            let start_time = std::time::Instant::now();
+            let x = sampling::euler_sample(
+                &mmdit,
+                &y,
+                &context,
+                num_inference_steps,
+                cfg_scale,
+                time_shift,
+                height,
+                width,
+            )?;
+            let dt = start_time.elapsed().as_secs_f32();
+            println!(
+                "Sampling done. {num_inference_steps} steps. {:.2}s. Average rate: {:.2} iter/s",
+                dt,
+                num_inference_steps as f32 / dt
+            );
+            x
+        };
+
+        {
+            let vb_vae = vb_fp16.rename_f(sd3_vae_vb_rename).pp("first_stage_model");
+            let autoencoder = build_sd3_vae_autoencoder(vb_vae)?;
+
+            // Apply TAESD3 scale factor. Seems to be significantly improving the quality of the image.
+            // https://github.com/comfyanonymous/ComfyUI/blob/3c60ecd7a83da43d694e26a77ca6b93106891251/nodes.py#L721-L723
+            autoencoder.decode(&((x / 1.5305)? + 0.0609)?)?
+        }
```

Comment:
Good point, I've pushed some changes to avoid a bit of the duplication (so the vae application and creation as well as the text model/mmdit application)